src/SpectatorView.Unity/Assets/SpectatorView/Scripts/StateSynchronization/NetworkedComponents/TextMeshPro/TextMeshProObserverBase.cs

```diff
@@ -119,8 +121,19 @@ public override void Read(SocketEndpoint sendingEndpoint, BinaryReader message)
                 TextMeshObserver.verticalMapping = (TextureMappingOptions)message.ReadByte();
                 TextMeshObserver.wordWrappingRatios = message.ReadSingle();
                 TextMeshObserver.wordSpacing = message.ReadSingle();
+
+                needsUpdate = true;

```

Comment:
When is this method called, relative to Update, and why can't we do the ForceMeshUpdate() right here?

Reply:
For whatever reason it doesn't resolve the solution. We have also tried changing to the execution order for component observers, but attempting to force an update on the same update pass the text mesh pro text is created doesnt result in the font resolving correctly/things rendering correctly.

Reply:
Hm. It's just... if that's something that happens after the update pass, we won't get the result before the next frame anyway. In that case delaying it with Invoke() could have a benefit that the class won't have an Update() method (so the reflection won't pick it up, so it won't be participating in the update pass every frame, even if there is nothing to update). But that's probably not a big deal.